src/inmanta/plugins.py

```diff
@@ -259,188 +391,268 @@ def __init__(self, namespace: Namespace) -> None:
 
     def normalize(self) -> None:
         self.resolver = self.namespace
-        self.argtypes = [self.to_type(arg.arg_type, self.namespace) for arg in self.arguments]
-        self.returntype = self.to_type(self._return, self.namespace)
 
-    def _load_signature(self, function: Callable[..., object]) -> List[PluginArgument]:
+        # Resolve all the types that we expect to receive as input of our plugin
+        for arg in self.kwargs.values():
+            # Normalizing the kwargs will also normalize the args as they are all
+            # included in this dict
```

Comment:
Could you add a comment about this where the attribute is declared? When I first read it I assumed `kwargs` would hold only the keyword-only arguments.

Reply:
I tried to make it more clear with an additional attribute: `all_args`